proposals/0000-extension-lifecycle-framework.md

```diff
@@ -0,0 +1,142 @@
+   * When an extension is deprecated, the user's guide and the compiler warning _must_ include a statement about the longevity of the extension, though this need not necessarily commit to a concrete time for removal. Additionally, the user's guide and the warning should direct users to information about migrating away from the deprecated extension.
+   * Any extension will be deprecated prior to removal.
+   * Some extensions in this category might be `OverlappingInstances`, `Rank2Types`, `RecordPuns`, and `TypeInType`.
+
+ * `Legacy` extensions are explicitly not recommended for new code, much like `Deprecated`, however, they are expected to be supported indefinitely. This may be to maintain an older language edition, maintain a bridge to a newer feature, or otherwise keep backward compatibility. Extensions that are `Legacy` may have some conflict with `Stable` extensions and explicitly do not need to be considered for an extension to be `Stable`. Some extensions in this category might be `CUSKs`, `DeepSubsumption`,  and `NPlusKPatterns`.
+
+Above we list expected categorizations of several extensions based on statements in the user's guide. We do not attempt to prescribe any particular categorizations as part of this proposal and simply provide a small number of expectations from the perspective of someone reading the existing documentation.
+
+The following transitions are included:
+ * (does not exist) -> `Experimental`
+ * `Experimental` -> `Stable`
+ * `Experimental` -> `Deprecated`
+ * `Experimental` -> `Legacy`
+ * `Stable` -> `Deprecated`
+ * `Stable` -> `Legacy`
+ * `Legacy` -> `Deprecated`
+ * `Deprecated` -> (does not exist)
+
+These transitions are explicitly excluded:
+ * `Stable` -> (does not exist)
+ * `Legacy` -> (does not exist)
+
+This is a small restriction, such that `Stable` or `Legacy` extensions require a deprecation cycle prior to removal.
+
+The following transitions are possible but left at the discretion of implementors as to be included or not:
+
+ * (does not exist) -> `Deprecated`
+ * (does not exist) -> `Legacy`
+ * (does not exist) -> `Stable`
+ * `Experimental` -> (does not exist)
+ * `Stable` -> `Experimental`
+ * `Legacy` -> `Experimental`
+ * `Legacy` -> `Stable`
+ * `Deprecated` -> `Experimental`
+ * `Deprecated` -> `Stable`
+ * `Deprecated` -> `Legacy`
+
+However, it also seems plausible that new knowledge might from time to time cause a stable extension to once again be considered experimental, e.g. in the face of soundness bugs or subtle interactions with other features. We explicitly allow for more transitions to allow for unforeseen circumstances. Restrictions are places on removal only. With this we aim to balance predictibility and flexibility. The consideration of both is important as extension classifications are first and foremost a communications channel between language designers, compiler developers, and users.
```

Comment:
```suggestion
However, it also seems plausible that new knowledge might from time to time cause a stable extension to once again be considered experimental, e.g. in the face of soundness bugs or subtle interactions with other features. We explicitly allow for more transitions to allow for unforeseen circumstances. Restrictions are placed on removal only. With this we aim to balance predictibility and flexibility. The consideration of both is important as extension classifications are first and foremost a communications channel between language designers, compiler developers, and users.
```

Reply:
Thanks again, this one should also be fixed now.